packages/api/src/command/webhook/fitbit.ts

```diff
@@ -32,80 +32,93 @@ const log = Util.log(`Fitbit Webhook`);
 export const processData = async (data: FitbitWebhook): Promise<void> => {
   console.log("Starting to process a Fitbit webhook.");
 
+  // Group notifications by Fitbit user ID
+  const fitbitUserGroupedNotifications: {
+    [fitbitUserId: string]: { collectionType: FitbitCollectionTypes; date: string }[];
+  } = {};
+
+  data.forEach(d => {
+    if (!fitbitUserGroupedNotifications[d.ownerId]) {
+      fitbitUserGroupedNotifications[d.ownerId] = [];
+    }
+    fitbitUserGroupedNotifications[d.ownerId].push({
+      collectionType: d.collectionType,
+      date: d.date,
+    });
+  });
+
+  // collect and group data by ConnectedUser
   const connectedUsersAndData = await Promise.all(
-    data.map(async d => {
-      const { collectionType, date, ownerId: fitbitUserId } = d;
-      try {
-        const connectedUsers = await getConnectedUsersByTokenOrFail(
-          ProviderSource.fitbit,
-          fitbitUserId
-        );
+    Object.entries(fitbitUserGroupedNotifications).map(async ([fitbitUserId, notifications]) => {
+      const connectedUser = await getConnectedUserByTokenOrFail(
+        ProviderSource.fitbit,
+        fitbitUserId
+      );
+      const accessToken = await Constants.PROVIDER_OAUTH2_MAP[ProviderSource.fitbit].getAccessToken(
+        connectedUser
+      );
 
-        const userFitbitData = await mapData(collectionType, connectedUsers[0], date);
-        const returnData: Entry[] = [];
+      const userFitbitData: WebhookUserDataPayload = {};
+      await Promise.all(
```

Comment:
Should we move to `Promise.allSettled` so we have better control over what executed and what failed? Here and on line 51.

Reply:
Okay. I was deliberately using `.all()` to reject as soon as something goes wrong. What's the thinking to use `allSettled()` instead?